lib/xcribe/request/validator.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Xcribe.Request.Validator do
+  alias Xcribe.Request
+  alias Xcribe.Request.Error
+
+  def validate(%Request{} = request) do
+    {:ok, request}
+    |> validate_parameters_in(:request_body)
+    |> validate_parameters_in(:path_params)
+    |> validate_parameters_in(:header_params)
+    |> validate_parameters_in(:query_params)
+    |> handle_validate(request)
+  end
+
+  defp validate_parameters_in({:error, _err} = error, _key), do: error
+
+  defp validate_parameters_in({:ok, request}, key) do
+    request
+    |> Map.fetch!(key)
+    |> find_struct()
+    |> handle_validate_params(request)
+  end
+
+  def find_struct(%{__struct__: module}) do
+    %Error{
+      type: :validation,
+      message: "The Plug.Conn params must be valid HTTP params. A struct #{module} was found!"
```

Comment:
@Danielwsx64 Would be possible to include the path to the struct found in the params? Sometimes is hard to find the place where the struct is been returned.

Would make sense to include it on xcribe, or is better to let the user debug the test to find it?

Reply:
I believe the request params are built in test setup section. It supposed to be easy find out where you added an struct to params. Any way I can't figure out how to point the path for the struct. If you have a suggestion for a easy way to do that we could add.

Reply:
> If you have a suggestion for a easy way to do that we could add.

An easy way, I don't have. I think we could accumulate the step took for each iteration in the recursion. 

But thinking better, seems to be a little out of scope of xcribe to provide this path.